### PR TITLE
Add Mark as read button for every unread notification

### LINKED ISSUE #1122: Add 'Mark as Read' button for every unread notification
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

**Current behavior:**
A common Mark all read button appears in the notification route. 
image

whereas in eventyay.com/notifications every unread notification too has 'Mark as read' button.
image

**Expected behavior:**
Every individual unread notification should have a mark as read button.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by ritikamotwani**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Mark as read button appears on every unread notification as on eventyay.com

#### Changes proposed in this pull request:

- Currently there is a common mark as read button which when clicked marks all notifications as read.
- There should be such a button on every individual notification too.
image

Fixes #1122

### CODE REVIEW

File: app/routes/notifications.js [hunk 1]
```diff
@@ -23,6 +23,16 @@ export default Route.extend(AuthenticatedRouteMixin, {
         .catch(() => {
           this.get('notify').error(this.get('l10n').t('An unexpected error occurred.'));
         });
+    },
+    markAsRead(notification) {
```
Comment (niranjan94) [inline app/routes/notifications.js]: This shouldn't be in the route. Move all actions to the controller please

**Comment by ritikamotwani** [comment]
@niranjan94 I have moved the action markAllRead to the controller but if I move the other one markAsRead which marks the individual notifications as read in the route /notifications the button when clicked shows an error:
image
whereas when the action is in the route it properly executes the buttons functionality.

**Comment by CosmicCoder96** [comment]
@ritikamotwani That's because you are most certainly incorrectly passing the action from the controller to the routes.  See other cases in the app's codebase how the action needs to be passed.

**Comment by ritikamotwani** [comment]
@CosmicCoder96 I think it is because the action is being called in all.hbs and the action has to be handled in two routes /notifications and /notifications/all so the action defined in the controller all.js is executing the action properly(route: notifications/all) but the controller notifications.js(for route: /notifications) is showing this error because of the template and controller names being different.

**Comment by ritikamotwani** [comment]
@niranjan94 @CosmicCoder96 @geekyd please review.

File: app/controllers/notifications/index.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import Controller from '@ember/controller';
+
+export default Controller.extend({
+  actions: {
+    markAsRead(notification) {
```
Comment (niranjan94) [inline app/controllers/notifications/index.js]: DRY. Take this action (and the one in `notifications/all`) into a common mixin.

**Comment by ritikamotwani** [comment]
@niranjan94 please review.

**Pull request merged**